pkgs/tools/misc/edk2-uefi-shell/default.nix

```
@@ -0,0 +1,33 @@
+{ lib
+, edk2
+, util-linux
+, nasm
+, python3
+}:
+edk2.mkDerivation "ShellPkg/ShellPkg.dsc" (finalAttrs: {
+  pname = "edk2-uefi-shell";
+  inherit (edk2) version;
+
+  nativeBuildInputs = [ util-linux nasm python3 ];
+  strictDeps = true;
+
+  # We only have a .efi file in $out which shouldn't be patched or stripped
+  dontPatchELF = true;
+  dontStrip = true;
+
+  # GUID hardcoded to match ShellPkg.dsc
+  installPhase = ''
+    runHook preInstall
+    install -D -m0644 Build/Shell/RELEASE*/*/Shell_EA4BB293-2D7F-4456-A681-1F22F42CD0BC.efi $out/shell.efi
+    runHook postInstall
+  '';
+
+  passthru.efi = "${finalAttrs.finalPackage}/shell.efi";
```

```suggestion
  passthru.efi = "${placeholder "out"}/shell.efi";
```
That should remove the need to the more expensive finalAttrs.

---

Oh no, this evaluates but doesn't actually work. :(

With placeholder:
```
$ nix eval --raw .#edk2-uefi-shell.efi -L
/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9/shell.efi⏎  
```

With `finalAttrs.finalPackage`:
```
$ nix eval --raw .#edk2-uefi-shell.efi -L
/nix/store/rcm0l534pppvp0b1n31ymflsab2wdk4d-edk2-uefi-shell-202202/shell.efi
```